internal/v1/handler_compose_image.go

```diff
@@ -74,7 +78,8 @@ func (h *Handlers) handleCommonCompose(ctx echo.Context, composeRequest ComposeR
 	}
 
 	var repositories []composer.Repository
-	if composeRequest.ImageRequests[0].SnapshotDate != nil {
+	// We should only build repository snapshots in the case where there is a snapshot date and no content template
+	if composeRequest.ImageRequests[0].SnapshotDate != nil && composeRequest.ImageRequests[0].ContentTemplate == nil {
```

Comment:
I think a condition needs to be added here to handle a non-nil content template:

1. content template is set => all repositories should come from the template;
2. else if snapshotdate is set => the snapshots should be resolved like is already the case;
3. else the default behaviour.

Reply:
pushed up a change to set RH repos here, doing it outside of buildCustomizations (since we're not touching those here) but using the same buildTemplateRepositories method